native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/NativeImagePlugin.java

```diff
@@ -144,19 +144,25 @@ private void configureJavaProject(Project project, Provider<NativeImageService>
             runTask.configure(run -> configureAgent(project, agent, buildExtension, runTask, processAgentFiles));
         });
 
+
+        // Testing part begins here. -------------------------------------------
+
         // In future Gradle releases this becomes a proper DirectoryProperty
         File testResultsDir = GradleUtils.getJavaPluginConvention(project).getTestResultsDir();
         DirectoryProperty testListDirectory = project.getObjects().directoryProperty();
 
-        // Testing part begins here.
+        // Add DSL extension for testing
+        NativeImageOptions testExtension = createTestExtension(project, buildExtension, testResultsDir);
+
         TaskCollection<Test> testTask = findTestTask(project);
         Provider<Boolean> testAgent = agentPropertyOverride(project, testExtension);
         TaskProvider<ProcessGeneratedGraalResourceFiles> processAgentTestFiles = registerProcessAgentFilesTask(project, PROCESS_AGENT_TEST_RESOURCES_TASK_NAME);
 
         testTask.configureEach(test -> {
             testListDirectory.set(new File(testResultsDir, test.getName() + "/testlist"));
             test.getOutputs().dir(testResultsDir);
-            test.systemProperty("graalvm.testids.outputdir", testListDirectory.getAsFile().get());
+            // Set system property read by the UniqueIdTrackingListener.
+            test.systemProperty("junit.platform.listeners.uid.tracking.output.dir", testListDirectory.getAsFile().get());
```

Comment:
This is also where we should register an action to cleanup the previously executed code. Something like:

```
test.doFirst("cleanup test ids", task -> {
                File dir = testListDirectory.getAsFile().get();
                if (dir.exists()) {
                    GFileUtils.deleteDirectory(dir);
                }
            });
```